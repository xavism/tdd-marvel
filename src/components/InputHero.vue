<template>
    <form @submit.prevent="$emit('submitted', name)">
        <input type="text" v-model="name">
        <button></button>
    </form>
</template>

<script>
export default {
  name: 'InputHero',
  data() {
    return {
      name: '',
    };
  },
};
</script>
