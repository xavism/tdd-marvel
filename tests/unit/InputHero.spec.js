import { shallowMount, createLocalVue } from '@vue/test-utils';
import InputHero from '@/components/InputHero';

const localVue = createLocalVue();

describe('InputHero', () => {
  const build = () => {
    const options = { localVue };
    const wrapper = shallowMount(InputHero, options);

    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button'),
    };
  };

  it('renders the component', () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main child components', () => {
    // arrange
    const { input, button } = build();

    // assert
    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it('calls "submitted" event when submitting form', () => {
    // arrange
    const expectedHero = 'Thor';
    const { wrapper, button, input } = build();
    input().element.value = expectedHero;

    // act
    input().trigger('input');
    button().trigger('click');
    button().trigger('submit');

    // assert
    expect(wrapper.emitted().submitted[0]).toEqual([expectedHero]);
  });
});
