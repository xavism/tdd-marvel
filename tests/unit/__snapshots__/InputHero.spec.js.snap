// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHero renders the component 1`] = `<form><input type="text"> <button></button></form>`;
